packages/rrweb/src/record/observers/canvas/canvas-manager.ts

```diff
@@ -227,16 +375,19 @@ export class CanvasManager implements CanvasManagerInterface {
         lastSnapshotTime &&
         timestamp - lastSnapshotTime < timeBetweenSnapshots
       ) {
-        rafId = onRequestAnimationFrame(takeCanvasSnapshots);
+        rafId = requestAnimationFrame(takeCanvasSnapshots);
         return;
       }
       lastSnapshotTime = timestamp;
 
-      getCanvas().forEach((canvas: HTMLCanvasElement) => {
+      getCanvas(canvasElement).forEach((canvas: HTMLCanvasElement) => {
         const id = this.mirror.getId(canvas);
         if (snapshotInProgressMap.get(id)) return;
         snapshotInProgressMap.set(id, true);
-        if (['webgl', 'webgl2'].includes((canvas as ICanvas).__context)) {
+        if (
+          !manualSnapshot &&
```

Comment:
We can move the `manualSnapshot` check outside, so that `getCanvas().forEach()` doesn't run at all since it looks like there is nothing to be done for manual snapshots.

Reply:
`createImageBitmap` is run for manual snapshots and it's nested inside `getCanvas().forEach()`

Reply:
Ah oops, was hard to see that in GH 👍